pkg/skaffold/util/tar.go

```diff
@@ -48,21 +51,55 @@ func CreateTar(w io.Writer, root string, paths []string) error {
 	defer tw.Close()
 
 	for _, path := range paths {
-		if err := addFileToTar(root, path, "", tw); err != nil {
+		if err := addFileToTar(root, path, "", tw, nil); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func CreateTarWithParents(w io.Writer, root string, paths []string, uid, gid int, modTime time.Time) error {
+	headerModifier := func(header *tar.Header) {
+		header.ModTime = modTime
+		header.Uid = uid
+		header.Gid = gid
+		header.Uname = ""
+		header.Gname = ""
+	}
+
+	tw := tar.NewWriter(w)
+	defer tw.Close()
+
+	// Make sure parent folders are added before files
+	// TODO(dgageot): this should probably also be done in CreateTar
+	// but I'd rather not break things that people didn't complain about!
+	added := map[string]bool{}
```

Comment:
this is interesting - it might be that (unless like here, you need the parents to have a certain uid/guid/modtime) TAR doesn't need parent folders to be created. Just "filespec, filecontent" serially. If this is the case then we can remove the TODO.